src/bin/pg_autoctl/pgctl.c

```diff
@@ -138,23 +138,25 @@ pg_controldata(PostgresSetup *pgSetup, bool missing_ok)
 	}
 	else
 	{
-		if (!missing_ok)
-		{
-			char *errorLines[BUFSIZE];
-			int lineCount = splitLines(prog.stderr, errorLines, BUFSIZE);
-			int lineNumber = 0;
+		char *errorLines[BUFSIZE];
+		int lineCount = splitLines(prog.stderr, errorLines, BUFSIZE);
```

Comment:
I know it is an earlier code. but splitLines function is not safe. There is no protection against writing  beyond buffer.

Reply:
I added the needed buffer size protection in splitLines.